nemo/collections/multimodal/speech_llm/data/lhotse_dataset.py

```diff
@@ -0,0 +1,493 @@
+import copy
+import random
+from typing import Optional
+
+import numpy as np
+import torch.utils.data
+from lhotse.dataset.collation import collate_vectors as collate_vectors_lhotse
+from nemo.utils import logging
+
+
+def collate_vectors(items, max_length: int, padding_value):
+    vectors = collate_vectors_lhotse(items, padding_value=padding_value)
+    if max_length > vectors.size(1):
+        vectors = torch.cat(
+            [vectors, padding_value * torch.ones(vectors.size(0), max_length - vectors.size(1), dtype=vectors.dtype)],
+            dim=1,
+        )
+    if items[0].shape[0] < 1:
+        vectors = vectors.long()
+    return vectors
+
+
+def ceil_to_nearest(n, m):
+    return (n + m - 1) // m * m
+
+
+class TextProcessing:
```

Comment:
I am not sure whether making lhotse dataloader part inherits nemo mtron dataloader is a good idea for future since we have decided to move away from nemo one at some point. Added docstring of the cls. Hope it helps.

Reply:
when have we decided to move away from nemo? If they share most of the code, we can move the TextProcessing class to another file like data_utils.py so that both versions import from the same file

Reply:
Done